pytx/pytx/request.py

```diff
@@ -0,0 +1,277 @@
+    def is_timestamp(timestamp):
+        """
+        Verifies the timestamp provided is a valid timestamp.
+
+        :param timestamp: Value to verify is a timestamp.
+        :type timestamp: str
+        :returns: True if valid, :class:`pytxValueError` if invalid.
+        """
+
+        try:
+            int(timestamp)
+            return True
+        except ValueError, e:
+            raise pytxValueError(e)
+
+    @staticmethod
+    def validate_limit(limit):
+        """
+        Verifies the limit provided is valid and within the max limit Facebook
+        will allow you to use (currently 5000).
+
+        :param limit: Value to verify is a valid limit.
+        :type limit: int, str
+        :returns: :class:`pytxValueError` if invalid.
+        """
+
+        try:
+            int(limit)
+        except ValueError, e:
+            raise pytxValueError(e)
+        if limit > t.MAX_LIMIT:
+            raise pytxValueError(
+                "limit cannot exceed %s (default: %s)" % (t.MAX_LIMIT,
+                                                          t.DEFAULT_LIMIT)
+            )
+        return
+
+    @staticmethod
+    def sanitize_strict(strict_text):
+        """
+        If strict_text is provided, sanitize it.
+
+        'true' will be used if strict_text is in [True, 'true', 'True', 1].
+        'false' will be used if strict_text is in [False, 'false', 'False', 0].
+
+        If we receive any other value strict_text will be set to None and
+        ignored when building the GET request.
+
+        :param strict_text: The value to sanitize.
+        :type strict_text: bool, str, int
+        :returns: str, None
+        """
+
+        if strict_text in (True, 'true', 'True', 1):
+            strict = 'true'
+        elif strict_text in (False, 'false', 'False', 0):
+            strict = 'false'
+        else:
+            strict = None
+        return strict
+
+    @staticmethod
+    def handle_results(resp):
+        """
+        Handle the results of a request.
+
+        :param resp: The HTTP response.
+        :type resp: response object
+        :returns: dict (using json.loads())
+        """
+
+        if resp.status_code != 200:
+            raise pytxFetchError("Response code: %s: %s" % (resp.status_code,
+                                                            resp.text))
+        try:
+            results = json.loads(resp.text)
+        except:
+            raise pytxFetchError("Unable to convert response to JSON.")
+        return results
+
+    @classmethod
+    def validate_get(cls, limit, since, until):
+        """
+        Executes validation for the GET parameters: limit, since, until.
+
+        :param limit: The limit to validate.
+        :type limit: int, str
+        :param since: The since timestamp to validate.
+        :type since: str
+        :param until: The until timestamp to validate.
+        :type until: str
+        """
+
+        if since:
+            cls.is_timestamp(since)
```

Comment:
The since and until parameters don't have to be timestamps/integers. There are some magic values, yesterday; today; and now, which can make searching easier.